pkg/platform/kube/scaler/poller.go

```diff
@@ -0,0 +1,149 @@
+package scaler
+
+import (
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/errors"
+
+	"github.com/nuclio/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type metricsPoller struct {
+	logger       logger.Logger
+	scaler       *Scaler
+	statsChannel chan metricEntry
```

Comment:
decoupling with a channel is ok, but limits what you can do. given my autoscaler simplification down to one go routine, do the following:
1. declare a `metricReporter` interface that has only `reportMetric`
2. add `reportMetric` to `autoscaler`. this sends a request to the channel with kind = metric - create this channel in the autoscaler channel, do not get it from the outside
3. pass `autoscaler` instance to poller as a `metricReporter` interface